src/test/java/de/tum/in/www1/artemis/exam/ExamParticipationIntegrationTest.java

```diff
@@ -0,0 +1,1340 @@
+
+    @Autowired
+    private SubmissionRepository submissionRepository;
+
+    @Autowired
+    private ResultRepository resultRepository;
+
+    @Autowired
+    private ParticipationTestRepository participationTestRepository;
+
+    @Autowired
+    private GradingScaleRepository gradingScaleRepository;
+
+    @Autowired
+    private TeamRepository teamRepository;
+
+    @Autowired
+    private BonusRepository bonusRepository;
+
+    @Autowired
+    private PlagiarismCaseRepository plagiarismCaseRepository;
+
+    @Autowired
+    private ParticipantScoreRepository participantScoreRepository;
+
+    @Autowired
+    private ProgrammingExerciseTestService programmingExerciseTestService;
+
+    @Autowired
+    private UserUtilService userUtilService;
+
+    @Autowired
+    private CourseUtilService courseUtilService;
+
+    @Autowired
+    private ExamUtilService examUtilService;
+
+    @Autowired
+    private TextExerciseUtilService textExerciseUtilService;
+
+    @Autowired
+    private ProgrammingExerciseUtilService programmingExerciseUtilService;
+
+    @Autowired
+    private ParticipationUtilService participationUtilService;
+
+    @Autowired
+    private TeamUtilService teamUtilService;
+
+    @Autowired
+    private GradingScaleUtilService gradingScaleUtilService;
+
+    private Course course1;
+
+    private Exam exam;
+
+    private static final int NUMBER_OF_STUDENTS = 3;
+
+    private static final int NUMBER_OF_TUTORS = 2;
+
+    private final List<LocalRepository> studentRepos = new ArrayList<>();
+
+    private User student1;
+
+    private User instructor;
+
+    @BeforeEach
+    void initTestCase() {
+        userUtilService.addUsers(TEST_PREFIX, NUMBER_OF_STUDENTS, NUMBER_OF_TUTORS, 0, 1);
+
+        course1 = courseUtilService.addEmptyCourse();
+        student1 = userUtilService.getUserByLogin(TEST_PREFIX + "student1");
+        instructor = userUtilService.getUserByLogin(TEST_PREFIX + "instructor1");
+
+        exam = examUtilService.addExamWithExerciseGroup(course1, true);
+        bitbucketRequestMockProvider.enableMockingOfRequests();
+
+        ParticipantScoreScheduleService.DEFAULT_WAITING_TIME_FOR_SCHEDULED_TASKS = 200;
+        participantScoreScheduleService.activate();
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        bitbucketRequestMockProvider.reset();
+        bambooRequestMockProvider.reset();
+        if (programmingExerciseTestService.exerciseRepo != null) {
+            programmingExerciseTestService.tearDown();
+        }
+
+        for (var repo : studentRepos) {
+            repo.resetLocalRepo();
+        }
+
+        ParticipantScoreScheduleService.DEFAULT_WAITING_TIME_FOR_SCHEDULED_TASKS = 500;
+        participantScoreScheduleService.shutdown();
+    }
+
+    // TODO IMPORTANT test more complex exam configurations (mixed exercise type, more variants and more registered students)
+    @Nested
+    class ExamStartTest {
```

Comment:
ExamParticipationIntegrationTest is still a very long class. We should extract this nested class (```ExamStartTest```) into its own file :)

Reply:
good idea, done ^^